### DISCUSSION TITLE
What's next after cmake install process?

### DISCUSSION THREAD

**Discussion by MikHaiLz404** [AcademySoftwareFoundation/OpenRV#73] [Build Issues]
Sorry everyone for a silly question, But I'm a fresh of code compile process.
At first I already got a build with a command below

`cmake --install cmake-build --config Release`

Also, after looking around in the target folder, I couldn't figure out how to run the open RV.
I have included folder structure that was created after the build for you to see. Please explain the next step to me.

Best

![open-rv](https://user-images.githubusercontent.com/42891902/213863753-853f7525-8e4a-4eff-9105-0ba95da10c40.jpg)

**Comment by marcomeyerVFX** [comment]
in the bin directory should be a `rv.exe` (I guess, I'm not on Windows) which you can run

**Comment by MikHaiLz404** [comment]
> in the bin directory should be a `rv.exe` (I guess, I'm not on Windows) which you can run

Thanks @marcomeyerVFX about the reply 😃 
I though the same as you but when I didn't find rv.exe or something might be related in the bin folder.
I sorted with executable file above the red line and attached the image to show you. 

![open-rv2](https://user-images.githubusercontent.com/42891902/213866028-e954db32-9583-4b67-a853-b3229d1b8f7d.jpg)

**Comment by RichardFrangenberg** [comment]
@MikHaiLz69 That means there were some errors during the build process. Unfortunately the build process doesn't stop then so it might be not obvious that there was an error.
You can write the whole output of the build process to a log file and then search it for errors: https://github.com/AcademySoftwareFoundation/OpenRV/issues/3#issuecomment-1397178404

**Comment by MikHaiLz404** [comment]
Thanks @RichardFrangenberg , I will check the log and find the error and fix it.

**Comment by BrianHanke** [comment]
I had a similar problem, convo here: https://github.com/AcademySoftwareFoundation/OpenRV/issues/3 

In the end I used an Azure VM to build OpenRV. There was something weird with the settings on my laptop so it was better just to start with a clean system.

**Comment by rajivfx** [comment]
I did compile open-rv and did not get any error. I am also facing same issue. No rv.exe file in bin folder.
Here what I used:

Window 11
Visual Studio 2019 With Window 10 and 11 SDK
Qt 5.15.2

Command: 
cmake -B cmake-build -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64
cmake --build cmake-build --config Release --target rv -j8 
cmake --install cmake-build --config Release

Log:

rajiv@zb MINGW64 /e/OpenRV/git_repo/OpenRV
# cmake -B cmake-build -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64
-- Build type: Release
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Building RV for Microsoft Windows
-- Using build branch: main
-- Using build hash: 7d4791d
-- RV_STAGE_ROOT_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app
-- RV_STAGE_BIN_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/bin
-- RV_STAGE_LIB_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/lib
-- RV_STAGE_INCLUDE_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/include
-- RV_STAGE_SRC_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/src
-- RV_STAGE_RESOURCES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/resources
-- RV_STAGE_PLUGINS_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns
-- RV_STAGE_PLUGINS_CONFIGFILES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/ConfigFiles
-- RV_STAGE_PLUGINS_IMAGEFORMATS_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/ImageFormats
-- RV_STAGE_PLUGINS_MOVIEFORMATS_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/MovieFormats
-- RV_STAGE_PLUGINS_MU_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Mu
-- RV_STAGE_PLUGINS_NODES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Nodes
-- RV_STAGE_PLUGINS_OIIO_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/OIIO
-- RV_STAGE_PLUGINS_PACKAGES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Packages
-- RV_STAGE_PLUGINS_PROFILES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Profiles
-- RV_STAGE_PLUGINS_PYTHON_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Python
-- RV_STAGE_PLUGINS_QT_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/Qt
-- RV_STAGE_PLUGINS_SUPPORTFILES_DIR: E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app/PlugIns/SupportFiles
-- Updating submodules
-- Copying Qt into E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app
-- Copying Qt translations files ...
-- Copying Qt resources files ...
-- Copying Qt into E:/OpenRV/git_repo/OpenRV/cmake-build/stage/app -- DONE
-- Populating rv_deps_opentimelineio
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Configuring done
-- Generating done
-- Build files have been written to: E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_opentimelineio-subbuild
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Performing update step for 'rv_deps_opentimelineio-populate'
  Building Custom Rule E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_opentimelineio-subbuild/CMakeLists.txt
  No patch step for 'rv_deps_opentimelineio-populate'
  No configure step for 'rv_deps_opentimelineio-populate'
  No build step for 'rv_deps_opentimelineio-populate'
  No install step for 'rv_deps_opentimelineio-populate'
  No test step for 'rv_deps_opentimelineio-populate'
  Completed 'rv_deps_opentimelineio-populate'
  Building Custom Rule E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_opentimelineio-subbuild/CMakeLists.txt
-- Populating rv_deps_pyside2
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22621.
-- Configuring done
-- Generating done
-- Build files have been written to: E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_pyside2-subbuild
Microsoft (R) Build Engine version 16.11.2+f32259642 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Performing update step for 'rv_deps_pyside2-populate'
  Building Custom Rule E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt
  No patch step for 'rv_deps_pyside2-populate'
  No configure step for 'rv_deps_pyside2-populate'
  No build step for 'rv_deps_pyside2-populate'
  No install step for 'rv_deps_pyside2-populate'
  No test step for 'rv_deps_pyside2-populate'
  Completed 'rv_deps_pyside2-populate'
  Building Custom Rule E:/OpenRV/git_repo/OpenRV/cmake-build/_deps/rv_deps_pyside2-subbuild/CMakeLists.txt
-- Using Boost:        1.76.0
-- Using Dav1d:        1.0.0
-- Using FFMPEG:       n4.4.3
-- Using atomic_ops:   7.7.0
-- Using Imath:        3.1.5
-- Using OpenEXR:      3.1.5
-- Using Python3:      3.9.15
-- Using PySide2:      5.15.2.1
-- Using Qt5:          5.15.2
-- Using zlib:         1.2.13
-- Adding "CMAKE_MINIMUM_REQUIRED(VERSION 3.24)" to INSTALL
-- Adding VARIABLE RV_APP_ROOT to INSTALL
-- Adding VARIABLE RV_BUILD_ROOT to INSTALL
-- Adding VARIABLE RV_CPP_STANDARD to INSTALL
-- Adding VARIABLE RV_C_STANDARD to INSTALL
-- Adding VARIABLE RV_DEPS_ATOMIC_OPS_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_BASE_DIR to INSTALL
-- Adding VARIABLE RV_DEPS_BOOST_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_DAV1D_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_DAVID_LIB_DIR to INSTALL
-- Adding VARIABLE RV_DEPS_DOWNLOAD_DIR to INSTALL
-- Adding VARIABLE RV_DEPS_FFMPEG_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_IMATH_CMAKE_DIR to INSTALL
-- Adding VARIABLE RV_DEPS_IMATH_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_LIST to INSTALL
-- Adding VARIABLE RV_DEPS_OPENEXR_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_OPENSSL_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_PYSIDE2_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_PYTHON3_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_QT5_LOCATION to INSTALL
-- Adding VARIABLE RV_DEPS_QT5_RESOURCES_FOLDER to INSTALL
-- Adding VARIABLE RV_DEPS_QT5_TRANSLATIONS_FOLDER to INSTALL
-- Adding VARIABLE RV_DEPS_QT5_VERSION to INSTALL
-- Adding VARIABLE RV_DEPS_WIN_PERL_ROOT to INSTALL
-- Adding VARIABLE RV_DEPS_ZLIB_INCLUDE_DIR to INSTALL
-- Adding VARIABLE RV_DEPS_ZLIB_VERSION to INSTALL
-- Adding VARIABLE RV_FLEX_MINOR_VERSION to INSTALL
-- Adding VARIABLE RV_GIT_BRANCH to INSTALL
-- Adding VARIABLE RV_GIT_COMMIT_SHORT_HASH to INSTALL
-- Adding VARIABLE RV_INTERNAL_APPLICATION_NAME to INSTALL
-- Adding VARIABLE RV_MAJOR_VERSION to INSTALL
-- Adding VARIABLE RV_PACKAGES_DIR to INSTALL
-- Adding VARIABLE RV_PACKAGE_LIST to INSTALL
-- Adding VARIABLE RV_RELEASE_DESCRIPTION to INSTALL
-- Adding VARIABLE RV_SHOW_ALL_VARIABLES to INSTALL
-- Adding VARIABLE RV_STAGE_BIN_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_INCLUDE_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_LIB_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_CONFIGFILES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_IMAGEFORMATS_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_MOVIEFORMATS_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_MU_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_NODES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_OIIO_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_PACKAGES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_PROFILES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_PYTHON_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_QT_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_PLUGINS_SUPPORTFILES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_RESOURCES_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_ROOT_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_SCRIPTS_DIR to INSTALL
-- Adding VARIABLE RV_STAGE_SRC_DIR to INSTALL
-- Adding VARIABLE RV_TARGET_STRING to INSTALL
-- Adding VARIABLE RV_TARGET_WINDOWS to INSTALL
-- Adding VARIABLE RV_VERSION_STRING to INSTALL
-- Adding VARIABLE RV_VERSION_YEAR to INSTALL
-- Adding SCRIPT E:/OpenRV/git_repo/OpenRV/cmake/install/pre_install.cmake to INSTALL
-- Adding SCRIPT E:/OpenRV/git_repo/OpenRV/cmake/install/pre_install_windows.cmake to INSTALL
-- Adding SCRIPT E:/OpenRV/git_repo/OpenRV/cmake/install/install_darwin.cmake to INSTALL
-- Adding SCRIPT E:/OpenRV/git_repo/OpenRV/cmake/install/post_install.cmake to INSTALL
-- Adding SCRIPT E:/OpenRV/git_repo/OpenRV/cmake/install/post_install_windows.cmake to INSTALL
-- Configuring done
-- Generating done
-- Build files have been written to: E:/OpenRV/git_repo/OpenRV/cmake-build

**Comment by rogernelson** [comment]
Hi @rajiv-hqvfx.  There are three commands there.  The first one configures cmake, the second one builds the dependencies and the third builds RV.

It looks like the output you pasted is coming from the configure command, which completed successfully.  Were there any errors from either of the two build commands? If the build did not complete successfully, the error is likely within one of these.

Note that if it is easier for you, you can do `source rvcmds.sh` from the root folder, followed by `rvmk` to do it all for you automatically (please ensure the paths that are printed in the shell from the source command are correct before doing `rvmk`).

**Comment by rajivfx** [comment]
I don't have JOM installed in my pc. Is this mandatory for build?
I found this build error:
  
jom: C:\openrv\OpenRV\cmake-build\_deps\rv_deps_pyside2-src\pyside3_build\py3.9-qt5.15.2-64bit-release\shiboken2\Makefile [all] Error 2
CUSTOMBUILD : error : Error compiling shiboken2 [C:\openrv\OpenRV\cmake-build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
  Traceback (most recent call last):
    File "C:\openrv\OpenRV\cmake-build\_deps\rv_deps_pyside2-src\setup.py", line 296, in <module>
      setup_runner.run_setup()
    File "C:\openrv\OpenRV\cmake-build\_deps\rv_deps_pyside2-src\build_scripts\setup_runner.py", line 168, in run_setup
      raise RuntimeError(msg)
  RuntimeError:
  setup.py invocation failed with exit code: 1.

**Comment by rogernelson** [comment]
Hi @rajiv-hqvfx.  Yes jom is needed to build.  If you don't have it installed, you can find instructions her in the Windows build instructions on how to do it : https://github.com/AcademySoftwareFoundation/OpenRV/blob/main/doc/build_system/config_windows.md

Specifically here:
>Note: You will also need jom, and it is included with Qt Creator (available from the Qt online installer). If you do not want to install Qt Creator, you can download it from [here](https://download.qt.io/official_releases/jom/) and copy the executable into the QT installation root directory under Tools/QtCreator/bin/jom.
